### PR TITLE
[Enhancement] List partition name to value encoding

### PR DESCRIPTION
**Pull request by Astralidea**
Fixes #issue

## What type of PR is this:
- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Checklist:
- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.1
  - [x] 3.0
  - [ ] 2.5
  - [ ] 2.4

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 10 / 11 (90.91%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/analyzer/AnalyzerUtils.java|10|11|90.91%|[847]|

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/AnalyzerUtils.java [hunk 1]
```diff
@@ -865,6 +866,24 @@ public static Map<String, AddPartitionClause> getAddPartitionClauseFromPartition
...
+    public static String getFormatPartitionValue(String value) {
+        StringBuilder sb = new StringBuilder();
+        for (int i = 0; i < value.length(); i++) {
+            char ch = value.charAt(i);
+            if ((ch >= 'A' && ch <= 'Z') || (ch >= 'a' && ch <= 'z') || (ch >= '0' && ch <= '9')) {
+                sb.append(ch);
+            } else if (ch == '-' || ch == ':' || ch == ' ') {
+                // remove;
+            } else {
+                int unicodeValue = value.codePointAt(i);
```
Comment (ABingHuang) [inline fe/fe-core/src/main/java/com/starrocks/sql/analyzer/AnalyzerUtils.java:879]: why not use unicode as partition name?
it may be easy to read.

Comment (ABingHuang) [reply fe/fe-core/src/main/java/com/starrocks/sql/analyzer/AnalyzerUtils.java:879]: it should be optimized later

**Review by ABingHuang — APPROVED** [review]

**Review by gengjun-git — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.0

**Pull request merged**